bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/checksum/DigestManager.java

```diff
@@ -40,10 +41,19 @@
 public abstract class DigestManager {
     private static final Logger logger = LoggerFactory.getLogger(DigestManager.class);
 
+    /*
+     * This ByteBufAllocator gives us (pooled) heap ByteBuf's that are backed by an accessible array.
+     * The backing array can be used to instantiate a protobuf ByteString without copying the array.
+     * This allocator should be used when using protobuf to eliminate array copies when converting
+     * ByteBuf to ByteString.
+     */
+    private static final ByteBufAllocator HEAP_BYTEBUF_ALLOCATOR = new PooledByteBufAllocator(false);
```

Comment:
I think you want need to provide here is an unpooled bytebuf allocator:

https://netty.io/4.0/api/io/netty/buffer/UnpooledByteBufAllocator.html#UnpooledByteBufAllocator-boolean-boolean-

Reply:
Yes. We can just call Unpooled.buffer() (same as if we instantiate UnpooledByteBufAllocator(false)).